src/java.base/share/classes/java/security/cert/X509Certificate.java

```diff
@@ -581,9 +585,16 @@ public X500Principal getSubjectX500Principal() {
      * in the form "a1:a2:...:a8", where a1-a8 are hexadecimal values
      * representing the eight 16-bit pieces of the address. OID names are
      * returned as {@code String}s represented as a series of nonnegative
-     * integers separated by periods. And directory names (distinguished names)
+     * integers separated by periods. Directory names (distinguished names)
      * are returned in <a href="http://www.ietf.org/rfc/rfc2253.txt">
-     * RFC 2253</a> string format. No standard string format is
+     * RFC 2253</a> string format. otherNames are returned as a byte array
+     * containing the ASN.1 DER encoded form of the name, and may also return
+     * a third entry in the list containing its {@code type-id} in string
+     * format, and a fourth entry as either a string (if {@code value} inside
+     * is a valid UTF8String, PrintableString, T61String, IA5String,
+     * UniversalString, BMPString, or GeneralString) or a byte array containing
+     * the ASN.1 DER encoding of {@code value} without the context-specific
+     * constructed tag with number 0 (otherwise). No standard string format is
      * defined for otherNames, X.400 names, EDI party names, or any
      * other type of names. They are returned as byte arrays
      * containing the ASN.1 DER encoded form of the name.
```

Comment:
I would move this sentence up before "otherNames ..." and remove the "otherNames" from it and the end of the sentence. So basically:

>      * No standard string format is
>      * defined for X.400 names or EDI party names. They are returned as byte arrays
>      * containing the ASN.1 DER encoded form of the name.
>      * otherNames are returned as a byte array ...